src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/DefaultLSPDocumentMappingProviderTest.cs

```diff
@@ -50,5 +64,220 @@ public async Task RazorMapToDocumentRangeAsync_InvokesLanguageServer()
+            var expectedVersion = 1;
+            var documentManager = new TestDocumentManager();
+            documentManager.AddDocument(RazorFile, Mock.Of<LSPDocumentSnapshot>(d => d.Version == expectedVersion && d.Uri == RazorFile));
+
+            var requestInvoker = GetRequestInvoker(new[]
+            {
+                ((RazorLanguageKind.CSharp, RazorFile, new TestRange(10, 10, 10, 15)), (expectedRange, expectedVersion))
+            });
+            var mappingProvider = new DefaultLSPDocumentMappingProvider(requestInvoker, documentManager);
+
+            var workspaceEdit = new TestWorkspaceEdit(versionedEdits: true);
+            workspaceEdit.AddEdits(RazorVirtualCSharpFile, 10, new TestTextEdit("newText", new TestRange(10, 10, 10, 15)));
+
+            // Act
+            var result = await mappingProvider.RemapWorkspaceEditAsync(workspaceEdit, CancellationToken.None).ConfigureAwait(false);
+
+            // Assert
+            var documentEdit = Assert.Single(result.DocumentChanges);
+            Assert.Equal(RazorFile, documentEdit.TextDocument.Uri);
+            Assert.Equal(expectedVersion, documentEdit.TextDocument.Version);
+
+            var actualEdit = Assert.Single(documentEdit.Edits);
+            Assert.Equal("newText", actualEdit.NewText);
+            Assert.Equal(expectedRange, actualEdit.Range);
+        }
+
+        [Fact]
+        public async Task RemapWorkspaceEditAsync_DocumentChangesNull_RemapsEditsAsExpected()
+        {
+            // Arrange
+            var expectedRange = new TestRange(1, 1, 1, 5);
+            var expectedVersion = 1;
+            var documentManager = new TestDocumentManager();
+            documentManager.AddDocument(RazorFile, Mock.Of<LSPDocumentSnapshot>(d => d.Version == expectedVersion && d.Uri == RazorFile));
+
+            var requestInvoker = GetRequestInvoker(new[]
+            {
+                ((RazorLanguageKind.CSharp, RazorFile, new TestRange(10, 10, 10, 15)), (expectedRange, expectedVersion))
+            });
+            var mappingProvider = new DefaultLSPDocumentMappingProvider(requestInvoker, documentManager);
+
+            var workspaceEdit = new TestWorkspaceEdit(versionedEdits: false);
+            workspaceEdit.AddEdits(RazorVirtualCSharpFile, 10, new TestTextEdit("newText", new TestRange(10, 10, 10, 15)));
+
+            // Act
+            var result = await mappingProvider.RemapWorkspaceEditAsync(workspaceEdit, CancellationToken.None).ConfigureAwait(false);
+
+            // Assert
+            Assert.Null(result.DocumentChanges);
+            var change = Assert.Single(result.Changes);
+            Assert.Equal(RazorFile.AbsoluteUri, change.Key);
+
+            var actualEdit = Assert.Single(change.Value);
+            Assert.Equal("newText", actualEdit.NewText);
+            Assert.Equal(expectedRange, actualEdit.Range);
+        }
+
+        [Fact]
+        public async Task RemapWorkspaceEditAsync_DoesNotRemapsNonRazorFiles()
```

Comment:
We should still be remapping other C# files shouldn't we?

Aka in FetchData.razor if you rename `WeatherForecast` the expectation is that the C# class outside of the Razor file will also be renamed.

Reply:
Are you confusing renaming with remapping? 😉

Reply:
Derp derp derp 👍